### ISSUE TITLE
S1764 should not report on Random.Next*

### ISSUE THREAD

**Issue by rudym87** [SonarSource/sonar-csharp#411]
### Description

Expressions are identical, but evaluated value are (almost always :wink:) not.

### Repro steps

````c#
var r = new Random();
if (r.NextDouble() > r.NextDouble())
{
    //...
}
````

### Expected behavior

No report.

### Actual behavior

False positive.

### Related information

* SonarC# version: 5.11.0.1761
* Sonar MSBuild Scanner version: 2.3.2.573

**Comment by Evangelink** [comment]
Hi @rudym87.

Thank you for your interest. We have been discussing this subject and even if you are right we still think that most of the time, when you write code with the same expression on both side of the operator there is a high probability you have done something wrong. Besides, as you probably know, you can ignore this particular issue on SonarQube.

So I am closing this ticket but feel free to keep discussing if you think we are missing something.
